src/main.rs

```diff
@@ -1,3 +1,116 @@
+// file-cryptor.exe -f -e -key <key>
+// file-cryptor.exe -f -d -key <key>
+
+use std::path::Path;
+use std::process;
+use std::{env, fs};
+
+struct Arguments {
+    file_name: String,
+
+    // true for encryption,
+    // false for decryption
+    mode: bool,
+    key: u8,
+}
+
+impl Arguments {
+    fn new(args: &[String]) -> Result<Arguments, &'static str> {
+        if args.len() != 6 {
+            return Err("invalid number of arguments.");
+        }
+
+        let file_name = Self::get_arg_value(args, "-f")
+            .ok_or("Invalid syntax. Expected -f in command arguments.")?;
+
+        if !file_name.ends_with(".txt") {
+            return Err("Invalid file name. Expected a .txt file.");
+        }
+
+        if !Path::new(&file_name).exists() {
+            return Err("File not found.");
+        }
+
+        let key_str = Self::get_arg_value(args, "-key")
+            .ok_or("Invalid syntax. Expected -key in command arguments.")?;
+
+        let key: u8 = match key_str.parse() {
+            Ok(k) => k,
+            Err(_) => return Err("invalid key format. Expected i32."),
+        };
+
+        let mode = if args.contains(&"-e".to_string()) {
+            true
+        } else if args.contains(&"-d".to_string()) {
+            false
+        } else {
+            return Err("invalid mode argument. Expected -e or -d.");
+        };
+
+        Ok(Arguments {
+            file_name,
+            mode,
+            key,
+        })
+    }
+
+    fn get_arg_value(args: &[String], flag: &str) -> Option<String> {
+        args.iter()
+            .position(|x| x == flag)
+            .and_then(|index| args.get(index + 1).cloned())
+    }
+}
+
+fn encryption(data: &mut Vec<u8>, key: u8) {
```

Comment:
1. Когда нужно конкретно индексироваться в буфере, передавать лучше слайсы: `&mut [u8]`. Vec к ним нормально приводится с zero copy
2. В данном случае лучше передавать итераторы, так как индексироваться вообще не нужно. Это позволит сделать например поточный шифр из условного файла:
```suggestion
fn encryption<I>(data: I, key: u8)
where
    I: Iterator<Item = &mut u8>,
{
```